### ISSUE TITLE
Consul ignoring catalog updates

### ISSUE THREAD

**Issue by foca** [hashicorp/consul#1283]
I'm trying to tie in consul (v0.5.2) and redis-sentinel so that whenever sentinel performs a failover, I switch the `master` tag accordingly.

Whenever the machines running sentinel perform a failover, they run [a script](https://gist.github.com/foca/2e7df8947346153379d5) that ideally would perform these updates.

Since sentinel does not run in the same nodes as the redis instances, I can't perform the update against the agent (since I can't specify the Node, or at least can't find in the API docs how to), so I'm updating the catalog directly.

When I perform the catalog update, consul responds with a 200, but querying the catalog afterwards tells me nothing changed.

The consul logs don't have any errors:

```
    2015/10/08 15:48:21 [DEBUG] http: Request /v1/catalog/service/redis (946.886µs)
    2015/10/08 15:48:21 [DEBUG] http: Request /v1/catalog/service/register (717.571µs)
    2015/10/08 15:48:21 [DEBUG] http: Request /v1/catalog/service/register (735.296µs)
```

I'm at a loss here at what do do. Any ideas of what could be going on / what I'm doing wrong / whether this is a bug are greatly appreciated :)

**Comment by slackpad** [comment]
Hi @foca - this is they key to what's going on:

> Since sentinel does not run in the same nodes as the redis instances, I can't perform the update against the agent (since I can't specify the Node, or at least can't find in the API docs how to), so I'm updating the catalog directly.

Currently, the agent is the source of truth so the next time anti-entropy sync runs it overwrites the changes you make to the catalog. Fortunately, there's a fix coming in 0.6 that will let you override tags (motivated by the Redis use case, actually) - see #1102. That should be released in a few weeks.

**Comment by foca** [comment]
Ah, cool. I'll figure out a workaround in the meantime. :+1: :purple_heart:
